src/main.rs

```diff
@@ -45,6 +46,18 @@ async fn main() -> miette::Result<()> {
             .await;
     }
 
+    for path in opts.read_logs_from {
+        manager
+            .spawn("read-logs", |handle| {
+                ReadLogsFrom {
+                    shutdown: handle,
+                    path,
+                }
+                .run()
```

Comment:
This is not a blocker, but it wasn't clear to me why this is split up as a struct and a method on that struct (i.e. `ReadLogsFrom { shutdown: handle, path }.run()`) and not combined into a single function (i.e. `readLogsFrom(handle, path)`)

Reply:
A couple reasons why I chose to implement it like this:

- Struct+method is a lot easier to split up to deal with complex logic.
- Methods don't have named parameters, so when there's a lot of state to pass in, using a struct is a lot clearer (and lower-overhead than defining builders).
- Most of the async tasks in `ghciwatch` are structured like this.